src/main/java/ai/knowly/langtorch/llm/minimax/MiniMaxService.java

```diff
@@ -0,0 +1,143 @@
+package ai.knowly.langtorch.llm.minimax;
+
+import ai.knowly.langtorch.llm.minimax.schema.MiniMaxApiExecutionException;
+import ai.knowly.langtorch.llm.minimax.schema.MiniMaxApiServiceInterruptedException;
+import ai.knowly.langtorch.llm.minimax.schema.config.MiniMaxServiceConfig;
+import ai.knowly.langtorch.llm.minimax.schema.dto.completion.ChatCompletionRequest;
+import ai.knowly.langtorch.llm.minimax.schema.dto.completion.ChatCompletionResult;
+import ai.knowly.langtorch.llm.minimax.schema.dto.embedding.EmbeddingRequest;
+import ai.knowly.langtorch.llm.minimax.schema.dto.embedding.EmbeddingResult;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.google.common.flogger.FluentLogger;
+import com.google.common.util.concurrent.ListenableFuture;
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import lombok.AllArgsConstructor;
+import okhttp3.ConnectionPool;
+import okhttp3.OkHttpClient;
+import retrofit2.Retrofit;
+import retrofit2.adapter.guava.GuavaCallAdapterFactory;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+/**
+ * @author maxiao
+ * @date 2023/06/07
+ */
+@AllArgsConstructor(access = lombok.AccessLevel.PRIVATE)
+public class MiniMaxService {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private static final String BASE_URL = "https://api.minimax.chat";
+
+  private final MiniMaxApi miniMaxApi;
+
+  public static MiniMaxService create(final MiniMaxApi api) {
+    return new MiniMaxService(api);
+  }
+
+  public static MiniMaxService create(final MiniMaxServiceConfig miniMaxServiceConfig) {
```

Comment:
Sorry. I understand that you are following the old style; however, we are switching to use Guice( a dependency injection framework) to decouple components. Please take a look at https://github.com/Knowly-ai/langtorch/issues/119

This means we would use less static factory method to do instance initialization (we had a lot of this before the switch) but going for default constructor that has @Inject annotation. Please update the code using latest OpenAIService as a reference. Thank you!!

Reply:
It's pretty a good idea, the code is more readability

had done this